packages/flutter/lib/src/material/color_scheme.dart

```diff
@@ -209,39 +418,113 @@ class ColorScheme with Diagnosticable {
   /// The color displayed most frequently across your app’s screens and components.
   final Color primary;
 
-  /// A darker version of the primary color.
-  final Color primaryVariant;
+  /// A color that's clearly legible when drawn on [primary].
+  ///
+  /// To ensure that an app is accessible, a contrast ratio of 4.5:1 for [primary]
+  /// and [onPrimary] is recommended. See
```

Comment:
suggestion for all of these: 'a contrast ratio between X and Y of at least 4.5:1 is recommended'